mlflow/gateway/providers/palm.py

```diff
@@ -0,0 +1,197 @@
+from typing import Any, Dict
+
+from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
+
+from mlflow.gateway.config import PaLMConfig, RouteConfig
+from mlflow.gateway.providers.base import BaseProvider
+from mlflow.gateway.providers.utils import rename_payload_keys, send_request
+from mlflow.gateway.schemas import chat, completions, embeddings
+
+
+class PaLMProvider(BaseProvider):
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, PaLMConfig):
+            raise TypeError(f"Unexpected config type {config.model.config}")
+        self.palm_config: PaLMConfig = config.model.config
+
+    async def _request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        headers = {"x-goog-api-key": self.palm_config.palm_api_key}
+        return await send_request(
+            headers=headers,
+            base_url="https://generativelanguage.googleapis.com/v1beta3/models/",
+            path=path,
+            payload=payload,
+        )
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        payload = jsonable_encoder(payload, exclude_none=True)
+        self.check_for_model_field(payload)
+        key_mapping = {
+            "stop": "stopSequences",
+            "candidate_count": "candidateCount",
+            "max_tokens": "maxOutputTokens",
```

Comment:
PaLM doesn't support this option for chat. Can we remove this casting and raise a 422 for the chat route if either "max_tokens" or "maxOutputTokens" is passed? 

Response if supplied:

```bash
HTTPError: 400 Client Error: Bad Request for url: http://127.0.0.1:8686/gateway/chat-palm/invocations. Response text: {"detail":"Invalid JSON payload received. Unknown name \"maxOutputTokens\": Cannot find field."}
```

Reply:
Additional info: https://developers.generativeai.google/api/rest/generativelanguage/models/generateMessage#request-body

Reply:
Sounds good added the check